Source/Core/UICommon/GameFileCache.cpp

```diff
@@ -0,0 +1,235 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "UICommon/GameFileCache.h"
+
+#include <algorithm>
+#include <cstddef>
+#include <functional>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "Common/ChunkFile.h"
+#include "Common/CommonTypes.h"
+#include "Common/File.h"
+#include "Common/FileSearch.h"
+#include "Common/FileUtil.h"
+
+#include "Core/TitleDatabase.h"
+
+#include "DiscIO/DirectoryBlob.h"
+
+#include "UICommon/GameFile.h"
+
+namespace UICommon
+{
+static constexpr u32 CACHE_REVISION = 7;  // Last changed in PR 6281
+
+std::vector<std::string> FindAllGamePaths(const std::vector<std::string>& directories_to_scan,
+                                          bool recursive_scan)
+{
+  static const std::vector<std::string> search_extensions = {
+      ".gcm", ".tgc", ".iso", ".ciso", ".gcz", ".wbfs", ".wad", ".dol", ".elf"};
+
+  // TODO: We could process paths iteratively as they are found
+  return Common::DoFileSearch(directories_to_scan, search_extensions, recursive_scan);
+}
+
+void GameFileCache::ForEach(std::function<void(const std::shared_ptr<const GameFile>&)> f) const
```

Comment:
This seems to be unused -- is that intended?

Reply:
It's used in GameListCtrl.cpp.